janitor/utils.py

```diff
@@ -539,69 +539,140 @@ def _grid_computation(entry: Dict) -> pd.DataFrame:
     return pd.DataFrame(*df_expand_grid, columns=df_columns)
 
 
-def _complete_groupings(df, list_of_columns):
-    # this collects all the columns as individual labels, which will be
-    # used to set the index of the dataframe
-    index_columns = []
-    # this will collect all the values associated with the respective
-    # columns, and used to reindex the dataframe, to get the complete
-    # pairings
-    reindex_columns = []
-    for item in list_of_columns:
-        if not isinstance(item, (str, dict, list, tuple)):
-            raise ValueError(
-                """Value must either be a column label, a list/tuple of columns or a
-                    dictionary where the keys are columns in the dataframe."""
-            )
-        if not item:
+# copied from pandas.core.common.py
+# used in _computations_complete
+# might also be useful in case_when
```

Comment:
@samukweku if this one is copied from the pandas' core, could we import it instead of copy/pasting it? Or is it modified?

Reply:
Thanks for the suggestion. Easier to import.